src/main/java/org/prebid/server/analytics/reporter/greenbids/GreenbidsAnalyticsReporter.java

```diff
@@ -0,0 +1,375 @@
+            auctionContext = auctionEvent.getAuctionContext();
+            bidResponse = auctionEvent.getBidResponse();
+        } else {
+            return Future.failedFuture(new PreBidException("Unprocessable event: " + event.getClass().getName()));
+        }
+
+        if (bidResponse == null || auctionContext == null) {
+            return Future.failedFuture(new PreBidException("Bid response or auction context cannot be null"));
+        }
+
+        final GreenbidsPrebidExt greenbidsBidRequestExt = parseBidRequestExt(auctionContext.getBidRequest());
+
+        final String greenbidsId = UUID.randomUUID().toString();
+        final String billingId = UUID.randomUUID().toString();
+
+        if (!isSampled(greenbidsBidRequestExt.getGreenbidsSampling(), greenbidsId)) {
+            return Future.succeededFuture();
+        }
+
+        final String commonMessageJson;
+        try {
+            final CommonMessage commonMessage = createBidMessage(
+                    auctionContext,
+                    bidResponse,
+                    greenbidsId,
+                    billingId,
+                    greenbidsBidRequestExt);
+            commonMessageJson = jacksonMapper.encodeToString(commonMessage);
+        } catch (PreBidException e) {
+            return Future.failedFuture(e);
+        } catch (EncodeException e) {
+            return Future.failedFuture(new PreBidException("Failed to encode as JSON: ", e));
+        }
+
+        final MultiMap headers = MultiMap.caseInsensitiveMultiMap()
+                .add(HttpUtil.ACCEPT_HEADER, HttpHeaderValues.APPLICATION_JSON)
+                .add(HttpUtil.CONTENT_TYPE_HEADER, HttpHeaderValues.APPLICATION_JSON);
+
+        final Future<HttpClientResponse> responseFuture = httpClient.post(
+                greenbidsAnalyticsProperties.getAnalyticsServerUrl(),
+                headers,
+                commonMessageJson,
+                greenbidsAnalyticsProperties.getTimeoutMs());
+
+        return processAnalyticServerResponse(responseFuture);
+    }
+
+    private GreenbidsPrebidExt parseBidRequestExt(BidRequest bidRequest) {
+        final Optional<ObjectNode> adapterNode = Optional.ofNullable(bidRequest)
+                .map(BidRequest::getExt)
+                .map(ExtRequest::getPrebid)
+                .map(ExtRequestPrebid::getAnalytics)
+                .filter(this::isNotEmptyObjectNode)
+                .map(analytics -> (ObjectNode) analytics.get(BID_REQUEST_ANALYTICS_EXTENSION_NAME));
+
+        final String pbuid = adapterNode.map(node -> node.get(PUBLISHER_ID_DYNAMIC_PARAM))
+                .map(JsonNode::asText)
+                .orElse(null);
+
+        final Double greenbidsSampling = adapterNode.map(node -> node.get(GREENBIDS_SAMPLING_DYNAMIC_PARAM))
+                .map(JsonNode::asDouble)
+                .orElse(null);
+
+        return GreenbidsPrebidExt.builder().pbuid(pbuid).greenbidsSampling(greenbidsSampling).build();
```

Comment:
I believe you can just convert the `adapterNode` to the GreenbidsPrebidExt with the jackson mapper